src/Hazelcast.Net.Tests/Caching/HazelcastCacheTests.cs

```diff
@@ -0,0 +1,496 @@
+                new("hazelcast:clusterName", RcCluster?.Id ?? ""),
+                new("hazelcast:caching:cacheUniqueIdentifier", cacheUniqueIdentifier),
+            })
+            .Build();
+
+        // add logging to the container, the normal way
+        services.AddLogging(builder => builder.AddConfiguration(configuration.GetSection("logging")).AddConsole());
+
+        // register Hazelcast options
+        services.AddHazelcastOptions(builder => builder
+
+            // get hazelcast options from the configuration defined above
+            .AddConfiguration(configuration) 
+
+            // override logger factory options to specify that it should be provided by the service provider
+            .With(options => options.ObtainLoggerFactoryFromServiceProvider())
+        );
+
+        // note: that would be a nicer syntax, but it requires that the factory has access
+        // back to the options (or some sort of IProvideServiceProvider service) and that
+        // would be a breaking change in the factory signature -> not now.
+        //options.LoggerFactory.ObtainFromServiceProvider();
+
+        // register Hazelcast cache options
+        // get them from the configuration defined above
+        services.Configure<HazelcastCacheOptions>(options => configuration.Bind("hazelcast:caching", options));
+
+        // add Hazelcast cache using provided options
+        services.AddHazelcastCache();
+
+        // the IDistributedCache will be injected into the asserter
+        services.AddTransient<ProvidedCacheAsserter>();
+
+        // create the provider and retrieve and use the cache
+        await using var serviceProvider = services.BuildServiceProvider();
+
+        await serviceProvider.GetRequiredService<ProvidedCacheAsserter>().AssertAsync();
+
+        // verifications
+        var ho = serviceProvider.GetRequiredService<IOptions<HazelcastOptions>>();
+        Assert.That(ho.Value.Networking.ReconnectMode, Is.EqualTo(ReconnectMode.ReconnectAsync));
+        var co = serviceProvider.GetRequiredService<IOptions<HazelcastCacheOptions>>();
+        Assert.That(co.Value.CacheUniqueIdentifier, Is.EqualTo(cacheUniqueIdentifier));
+
+        // note:
+        //   standard dotnet key delimiter is ':' or '__' depending on providers, but AddHazelcastAndDefaults add
+        //   support for '.' too for command line, environment and in-memory collection providers - but only for keys
+        //   starting with 'hazelcast.' or 'hazelcast-failover.'. Internally, the '.' separator is rewritten into the
+        //   ':' separator - which should be used everywhere really, for instance in the Bind call used in the
+        //   HazelcastCacheOptions registration. we're using the standard ':' delimiter everywhere in this test.
+    }
+
+    [Test]
+    [Category("enterprise")]
+    [Timeout(10000)]
+    public async Task TestCacheProvided_ConfiguredFailoverOptions()
```

Comment:
This test and TestCacheProvided_ConfiguredOptions; can't they use a common method to reduce duplicate code?

Reply:
totally, refactoring